.github/workflows/release.yaml

```diff
@@ -0,0 +1,62 @@
+---
+name: Publish releases
+
+# yamllint disable-line rule:truthy
+on:
+  release:
+    types: [published]
+env:
+  PYTHON_VERSION: "3.12"
+
+jobs:
+  build-and-publish-pypi:
+    name: Builds and publishes releases to PyPI
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4.1.7
+      - name: Get tag
+        id: vars
+        run: echo "tag=${GITHUB_REF#refs/*/}" >> $GITHUB_ENV
+      - name: Validate version number
+        run: >-
+          if [[ "${{ github.event.release.prerelease }}" == "true" ]]; then
+            if ! [[ "${tag}" =~ "b" ]]; then
+            echo "Pre-release: Tag is missing beta suffix (${tag})"
+              exit 1
+            fi
+          else
+            if [[ "${tag}" =~ "b" ]]; then
+              echo "Release: Tag must not have a beta suffix (${tag})"
+              exit 1
+            fi
+          fi
+      - name: Set up Python ${{ env.PYTHON_VERSION }}
+        uses: actions/setup-python@v5.1.1
+        with:
+          python-version: ${{ env.PYTHON_VERSION }}
+      - name: Install build
+        run: >-
+          pip install build tomli tomli-w
+      - name: Set Python project version from tag
+        shell: python
+        run: |-
+          import tomli
+          import tomli_w
+
+          with open("pyproject.toml", "rb") as f:
+            pyproject = tomli.load(f)
+
+          pyproject["project"]["version"] = "${tag}"
+
+          with open("pyproject.toml", "wb") as f:
+            tomli_w.dump(pyproject, f)
+      - name: Build python package
+        run: >-
+          python3 -m build
+      - name: Publish release to PyPI
+        uses: pypa/gh-action-pypi-publish@v1.9.0
+        with:
+          user: __token__
+          password: ${{ secrets.PYPI_TOKEN }}
```

Comment:
It's better to configure a trusted publisher at PyPI.

https://packaging.python.org/en/latest/guides/publishing-package-distribution-releases-using-github-actions-ci-cd-workflows/

Reply:
Thanks @MartinHjelmare , I didn't know about that. I was looking at python-matter-server as an example, seems we need to udpate that. I re-wrote using their guide. I also added an option to publish to TestPyPI as a workflow dispatch for testing before releases if we want. Let me know what you think.

Reply:
Looks great!